File: js/spec/test-board-datatxt.js
var pins = require('./helpers-pins-host').pins;
var test = require('./helpers-board-test');

describe('Board DataTxt', () => {
  const tempfile = pins.callbacks.get('tempfile');
  const board = 'simpletxt';
  const url = 'https://raw.githubusercontent.com/mlverse/pins/pins-js/js/spec/fixtures/datatxt/data.txt';

  it('can board_register() a datatxt board', async () => {
    await pins.boardRegister('datatxt', { name: board, url, cache: tempfile() });
    expect(pins.boardList().includes(board)).toBe(true);
  });

  it('can pin_get() iris from a datatxt board', async () => {
    const pin = await pins.pinGet('iris', { board });

    expect(pin.length).toBe(150);
    expect(pin[0].length).toBe(5);
  });

  it('can pin_find() mtcars_expr from a datatxt board', async () => {
    const result = await pins.pinFind('mtcars_expr', { board, metadata: true });

    expect(result.length).toBe(1);
    expect(result[0].metadata.rows).toBe(32);
    expect(result[0].metadata.cols).toBe(11);

    expect(typeof(result[0].metadata.rows)).toBe('number');
    expect(typeof(result[0].metadata.cols)).toBe('number');
  });

  it('can board_deregister() a datatxt board', () => {
    pins.boardDeregister(board);

    expect(pins.boardList().includes(board)).toBe(false);
  });

  it('can board_register() with URL and name', async () => {
    const boardName = await pins.boardRegister(url, { name: board, cache: tempfile() });

    expect(boardName).toBe(board);
    pins.boardDeregister(boardName);
  });

  it('can board_register() with URL and no name', async () => {
    const boardName = await pins.boardRegister(url, { cache: tempfile() });

    expect(boardName).toBe('raw');
  });

  it('can board_register() with URL, no name and data frame', async () => {
    const pinsPath = pins.options['pins.path'];

    pins.options['pins.path'] = tempfile();

    const irisPin = await pins.pinGet('iris', { board: url });

    expect(irisPin.length).toBe(150);

    pins.options['pins.path'] = pinsPath;
  });
});

File: python/tests/test_board_datatxt.py
import pins
import tempfile
import random
import os

board = 'simpletxt'
url = 'https://raw.githubusercontent.com/mlverse/pins/pins-js/js/spec/fixtures/datatxt/data.txt'

def test_register_datatxt():
  cache = os.path.join(tempfile.gettempdir(), str(random.randint(1, 100000)))
  pins.board_register('datatxt', { "name": board, "url": url, "cache": cache })

  assert board in pins.board_list()

def test_get_pin_datatxt():
  pin = pins.pin_get('iris', { "board": board })

  assert len(pin) == 151
  assert len(pin[0]) == 5

def test_find_pin_datatxt():
  pin = pins.pin_find('mtcars_expr', { "board": board, "metadata": True })

  print(pin)

  assert len(pin) == 1
  assert pin[0]["metadata"]["rows"] == 32
  assert pin[0]["metadata"]["cols"] == 11

  assert type(pin[0]["metadata"]["rows"]).__name__ == "int"
  assert type(pin[0]["metadata"]["cols"]).__name__ == "int"

def test_deregister_datatxt():
  pins.board_deregister(board)

  assert board not in pins.board_list()

def test_register_url_name_datatxt():
  cache = os.path.join(tempfile.gettempdir(), str(random.randint(1, 100000)))
  name = pins.board_register(url, { "name": board, "cache": cache })

  assert name == board

  pins.board_deregister(name)

def test_register_url_noname_datatxt():
  cache = os.path.join(tempfile.gettempdir(), str(random.randint(1, 100000)))
  name = pins.board_register(url, { "cache": cache })

  assert name == 'raw'

  pin = pins.pin_get('iris', { "board": url })

  assert len(pin) == 151
